backup: TestScheduledBackupCompaction failed

backup.TestScheduledBackupCompaction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19200728?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19200728?buildTab=artifacts#/) on master @ [7984c75b1fd8fafb35493e3b4de5140587a99452](https://github.com/cockroachdb/cockroach/commits/7984c75b1fd8fafb35493e3b4de5140587a99452):


```
=== RUN   TestScheduledBackupCompaction
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/9fe967a34c795ddf17fa3bc6249fb051/logTestScheduledBackupCompaction2306716223
    test_log_scope.go:76: use -show-logs to present logs inline
I250324 22:23:08.684959 1 (gostd) rand.go:256  [-] 1  random seed: 7870855692160395676
    backup_compaction_test.go:475: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9fe967a34c795ddf17fa3bc6249fb051/logTestScheduledBackupCompaction2306716223
--- FAIL: TestScheduledBackupCompaction (48.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScheduledBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48810

backup.TestScheduledBackupCompaction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19201426?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19201426?buildTab=artifacts#/) on master @ [7984c75b1fd8fafb35493e3b4de5140587a99452](https://github.com/cockroachdb/cockroach/commits/7984c75b1fd8fafb35493e3b4de5140587a99452):


```
=== RUN   TestScheduledBackupCompaction
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/9fe967a34c795ddf17fa3bc6249fb051/logTestScheduledBackupCompaction3810202413
    test_log_scope.go:76: use -show-logs to present logs inline
    backup_compaction_test.go:474: error scanning '&{<nil> 0x400194ab40}': sql: no rows in result set
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9fe967a34c795ddf17fa3bc6249fb051/logTestScheduledBackupCompaction3810202413
--- FAIL: TestScheduledBackupCompaction (6.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScheduledBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestScheduledBackupCompaction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19201572?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19201572?buildTab=artifacts#/) on master @ [7984c75b1fd8fafb35493e3b4de5140587a99452](https://github.com/cockroachdb/cockroach/commits/7984c75b1fd8fafb35493e3b4de5140587a99452):


```
=== RUN   TestScheduledBackupCompaction
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/9fe967a34c795ddf17fa3bc6249fb051/logTestScheduledBackupCompaction3812526955
    test_log_scope.go:76: use -show-logs to present logs inline
    backup_compaction_test.go:474: error scanning '&{<nil> 0x400647e000}': sql: no rows in result set
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9fe967a34c795ddf17fa3bc6249fb051/logTestScheduledBackupCompaction3812526955
--- FAIL: TestScheduledBackupCompaction (6.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScheduledBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/backup_compaction_test.go
+++ b/pkg/backup/backup_compaction_test.go
@@ -428,8 +428,6 @@ func TestScheduledBackupCompaction(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 143394, "flaky test")
-
 	ctx := context.Background()
 	th, cleanup := newTestHelper(t)
 	defer cleanup()
@@ -470,10 +468,16 @@ func TestScheduledBackupCompaction(t *testing.T) {
 	th.waitForSuccessfulScheduledJob(t, inc.ScheduleID())
 
 	var jobID jobspb.JobID
-	th.sqlDB.QueryRow(
-		t,
-		`SELECT job_id FROM [SHOW JOBS] WHERE description ILIKE 'COMPACT%' AND job_type = 'BACKUP'`,
-	).Scan(&jobID)
+	// The scheduler is notified of the backup job completion and then the
+	// compaction job is created in a separate transaction. As such, we need to
+	// poll for the compaction job to be created.
+	testutils.SucceedsSoon(t, func() error {
+		return th.sqlDB.DB.QueryRowContext(
+			ctx,
+			`SELECT job_id FROM [SHOW JOBS] WHERE description ILIKE 'COMPACT%' AND job_type = 'BACKUP'`,
+		).Scan(&jobID)
+	})
+
 	testutils.SucceedsSoon(t, func() error {
 		th.server.JobRegistry().(*jobs.Registry).TestingNudgeAdoptionQueue()
 		var unused int64

